modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java

```diff
@@ -528,6 +529,25 @@ public int getObjectDefinitionsCount(long companyId)
 		return objectDefinitionPersistence.countByCompanyId(companyId);
 	}
 
+	@Override
+	public ObjectDefinition getSystemObjectDefinition(
+			long companyId, String name)
+		throws PortalException {
+
+		for (ObjectDefinition systemObjectDefinition :
```

Comment:
I wonder if it would be better for performance to use:

```
	@Override
	public ObjectDefinition fetchObjectDefinition(long companyId, String name) {
		return objectDefinitionPersistence.fetchByC_N(companyId, name);
	}
```

and then, check if `objectDefinition != null && objectDefinition.isSystem()`

Reply:
I didn't want to use it as, because I have some doubts about the uniquess naming of system objects. I think that this `fetchObjectDefinition` only applies to custom objects. I based this theory in this test I made
![image](https://user-images.githubusercontent.com/17163902/225319017-e87fbd19-e6c0-4bf6-939e-060d34692b16.png)

As you can see, I am able to create an object with the exact naming that a system user. 

To be honest, I didn't test which ObjectDefinition will be returned by that method, I just assumed that it is only for custom objects and I used this other approximation 😅